src/views/v3/Bridge/AssetPicker/index.tsx

```diff
@@ -0,0 +1,557 @@
+    // If the amount input is empty, we don't need to check the quote which may be for the previous amount
+    if (!amount || amount.amount === '' || amount.amount === '0') {
+      return 0;
+    }
+    return props.quote
+      ? sdkAmount.whole(props.quote?.destinationToken.amount)
+      : undefined;
+  }, [amount, props.quote]);
+
+  const balance = useMemo(() => {
+    if (!props.isSource || !props.wallet.address) {
+      return null;
+    }
+
+    return (
+      <Typography color={theme.palette.text.secondary} variant="body2">
+        {tokenBalance
+          ? sdkAmount.display(sdkAmount.truncate(tokenBalance, 6))
+          : '0'}
+      </Typography>
+    );
+  }, [
+    props.isSource,
+    props.wallet.address,
+    tokenBalance,
+    theme.palette.text.secondary,
+  ]);
+
+  const handleAmountChange = useCallback((newValue: string): void => {
+    setAmountInput(newValue);
+  }, []);
+
+  const handleDebouncedAmountChange = useCallback(
+    (newValue: string): void => {
+      dispatch(setAmount(newValue));
+      setDebouncedAmountInput(newValue);
+    },
+    [dispatch],
+  );
+
+  // Clear the amount input value if the amount is reset outside of this component
+  // This can happen if user swaps selected source and destination assets.
+  useEffect(() => {
+    if (!amount && (amountInput || debouncedAmountInput)) {
+      handleAmountChange('');
+      handleDebouncedAmountChange('');
+    }
+    // We should run this sife-effect only when the amount changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [amount]);
+
+  const renderPercentButton = useCallback(
+    (percent: number) => (
+      <Button
+        sx={styles.percentButton}
+        disabled={props.isTransactionInProgress}
+        onClick={() => {
+          if (tokenBalance) {
+            const balancePercent =
+              (sdkAmount.units(tokenBalance) * BigInt(percent)) / BigInt(100);
+            const displayAmount = sdkAmount.display(
+              sdkAmount.fromBaseUnits(balancePercent, tokenBalance.decimals),
+            );
+            handleAmountChange(displayAmount);
+            handleDebouncedAmountChange(displayAmount);
+          }
+        }}
+      >
+        <Typography fontSize={12} fontWeight={500} textTransform="none">
+          {percent === 100 ? 'Max' : `${percent}%`}
+        </Typography>
+      </Button>
+    ),
+    [
+      handleAmountChange,
+      handleDebouncedAmountChange,
+      props.isTransactionInProgress,
+      tokenBalance,
+      styles.percentButton,
+    ],
+  );
+
+  const percentButtons = useMemo(() => {
+    if (!props.wallet.address || !tokenBalance) {
+      return null;
+    }
+
+    return (
+      <Box sx={{ display: 'flex', gap: '6px' }}>
+        {renderPercentButton(25)}
+        {renderPercentButton(50)}
+        {renderPercentButton(100)}
+      </Box>
+    );
+  }, [props.wallet.address, tokenBalance, renderPercentButton]);
+
+  return (
+    <Box sx={styles.root}>
```

Comment:
So I think this component can definitely be broken down quite a bit. If markup is going beyond 100 lines of code, I think it deserves to be extracted out so that when reading HTML one can mentally figure out how the structure is visually. And its lot easier to review for future changes. At the moment one has to scroll a lot just to understand what the structure is specially hard to figure out where ? : start and end.

Reply:
```jsx
<Box>
    <AssetPickerHeader>
    <AssetPickerContent>
</Box>
```